File: src/main/java/jromp/examples/BasicUsage.java
package jromp.examples;

import jromp.Constants;
import jromp.parallel.Parallel;

public class BasicUsage {
	public static void main(String[] args) {
		Parallel.defaultConfig()
		        .begin((id, variables) -> System.out.printf("Hello World from thread %d of %d%n", id,
		                                                    variables.<Integer>get(Constants.NUM_THREADS).get()))
		        .join();
	}
}

File: src/main/c/jromp/examples/basic_usage.c
#include <omp.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
    int threads;
    int tid;

    #pragma omp parallel private(tid)
    {
        tid = omp_get_thread_num();
        threads = omp_get_num_threads();
        printf("Hello World from thread %d of %d\n", tid, threads);
    }

    return 0;
}
